### ISSUE TITLE
Rule for parenthesis in multiline conditions

### ISSUE THREAD

**Issue by evilebottnawi** [eslint/eslint#4376]
First code style (parenthesis on line with end condition):
```javascript
var a = 1;
var b = 2;

if (a === 1
    && b === 2) { // <-- Here `)`
    // Code
}
```
Second code style (parenthesis on new line):
```javascript
var a = 1;
var b = 2;

if (a === 1
    && b === 2
) {     // <-- Here `)`
    // Code
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I'm sorry, I'm not sure I understand what this issue is? Is the reporting a bug, or requesting a new rule? If it's a bug, please answer the questions that bot asked in a follow-up, if it's a rule request, there's a link for Proposing a New Rule instructions in the bot's message.

**Comment by evilebottnawi** [comment]
@ilyavolodin requesting a new rule

**Comment by ilyavolodin** [comment]
In that case, please see this [link](http://eslint.org/docs/developer-guide/contributing/new-rules)

**Comment by gyandeeps** [comment]
@evilebottnawi Can you provide more information for your proposal? Here are the things we are looking for: http://eslint.org/docs/developer-guide/contributing/new-rules

**Comment by evilebottnawi** [comment]
@gyandeeps 
1. Modify rule `brace-style` for long conditions in parenthesis. Example (parenthesis on line with end condition and parenthesis on new line after all condition). 
Also example:
```javascript
if (conditionFirst === true && conditionSecond === true && conditionThird === true 
    && conditionN === true) {
    // Logic
}
```
```javascript
if (conditionFirst === true && conditionSecond === true && conditionThird === true 
    && conditionN === true
) {
    // Logic
}
```
I work in a small team and we use eslint for lint our code. Some team members adhere to the style of the first, the second adhere to the style of the second, it brings a different style of code. This would help to determine how to write correctly based on config for this rule.
2. Stylistic.
3. This would help make the code style much better and controlled. 
A similar rule for php (https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/PEAR/Sniffs/ControlStructures/MultiLineConditionSniff.php). 
4. Yes.

**Comment by nzakas** [comment]
I don't think it makes sense to modify brace-style for this. The braces should be fine so long as they appear in the correct location relative to the condition. It sounds like you're proposing a new rule. Are you willing to submit a PR for this?

**Comment by evilebottnawi** [comment]
@nzakas Yes

**Comment by mysticatea** [comment]
:+1:  I like this idea.

**Comment by nzakas** [comment]
Okay, let's give it a try.

**Comment by alberto** [comment]
@evilebottnawi do you still want to submit a PR?

**Comment by evilebottnawi** [comment]
@alberto no time

**Comment by alberto** [comment]
OK, closing then.

**Comment by mysticatea** [comment]
oh, may I make this later?
I want to use this.

**Comment by ilyavolodin** [comment]
@mysticatea Whenever you have time to work on this, either reopen this issue, or just create a new one pointing back to this and mark it as accepted.

**Comment by alberto** [comment]
Since we have tightened our requirements for new rules, up to @nzakas to decide if this would still be appropriate.

**Comment by evilebottnawi** [comment]
@alberto I'm ready to do this, it is better to make a separate rule? And what do you think about name for rule?

**Comment by mczepiel** [comment]
FWIW, I'd really prefer if you could just switch to `allman` style for multi-line conditionals, leaving the closing parent on the conditional instead of bringing it to the new line.

```
if (foo &&
  bar &&
  baz)
{
  body
}
```

Seems like it might be more acceptable to have the existing rule vary between already existing  strategies based on the conditional being a single or multi-line  as opposed to a new rule.

But yeah, I'm not sure how receptive everybody is to altering existing options and if you really insist on having the closing paren and the opening brace on the same line it looks like you'll need a new rule.

**Comment by mgcrea** [comment]
I think we should also handle non `if-related` cases, eg.

```
  const isModal = (location = this.props.location) => (
    location &&
    location.state &&
    location.state.modal
  )
```

enforced as

```
  const isModal = (location = this.props.location) => (
    location
      && location.state
      && location.state.modal
  )
```

Or is there another rule to target this specific case?
